lib/logger/lib/logger.ex

```diff
@@ -876,6 +876,8 @@ defmodule Logger do
 
   defguardp is_msg(msg) when is_binary(msg) or is_list(msg) or is_map(msg)
 
+  # TODO: Update :logger.macro_log/4 first argument type to include %{}
+  @dialyzer {:nowarn_function, __do_log__: 3}
```

Comment:
There are `logger:macro_log/4` calls with an empty map as first argument in the `logger` module. How are these not causing dialyzer issues for erlang itself?

Reply:
Try adding `generated: true` to the quote that emits `macro_log` and see if it addresses them.

Reply:
Sorry, I meant the erlang `:logger` module. I'll check how this gets through erlang's own dialyzer check on their CI

Reply:
This is Elixir calling Erlang logger through Erlang's undocumented interface that Erlang uses after expanding their macros. I was wrong before, Erlang is never calling these with an empty map as location so they don't get this dialyzer error.

I am not sure why we're not using the regular Erlang function-based logging interface or why we're not passing location information but there are probably important reasons for that.